app/dao/failed_logins_by_ip_dao.py

```diff
@@ -0,0 +1,33 @@
+import datetime
+
+from sqlalchemy import desc
+
+from app import db
+from app.models import FailedLoginCountByIP
+
+
+def dao_get_failed_logins():
+    return FailedLoginCountByIP.query.all()
+
+
+def dao_create_failed_login_for_ip(ip):
+    if FailedLoginCountByIP.query.filter_by(ip=ip).first():
+        latest_failed_login_count = dao_get_latest_failed_login_by_ip(ip).failed_login_count
+        data = FailedLoginCountByIP(
+            ip=ip,
+            failed_login_count=latest_failed_login_count + 1,
+            attempted_at=datetime.datetime.now(),
+        )
+    else:
+        data = FailedLoginCountByIP(
+            ip=ip,
+            failed_login_count=1,
+            attempted_at=datetime.datetime.now(),
+        )
```

Comment:
There are several race conditions here whereby simultaneous attempts will overwrite each other or fail to insert.

Ideally you should use an upsert statement to do both at the same time.  If your database doesn't support upsert (postgres and aurora both do) then at the very least your update statement should increment directly within the database rather than fetching a value then incrementing it and saving the changed value back to the database.

For the purposes of this example. I'm going to assume you've gotten rid of the ID column and made the IP column the primary key as per my comment below.

In SQL terms
```
INSERT INTO failed_logins (ip, failed_login_count, attempted_at)
VALUES ('127.0.0.1',1,NOW())
ON CONFLICT (ip)
DO UPDATE SET failed_login_count = failed_login_count+1;
```

If you didn't have an upsert command then you'd do the following
```
INSERT INTO failed_logins (ip, failed_login_count, attempted_at)
VALUES ('127.0.0.1',1,NOW())
```
If that insert fails (due to an existing IP already being there).
```
UPDATE failed_logins
SET failed_login_count = failed_login_count+1
WHERE ip='127.0.0.1'
```